index.js

```diff
@@ -381,6 +381,15 @@ function provider (registry, { Biome, version }) {
     return 0
   }
 
+  function getHashValue (states, name) {
+    const tag = nbt.comp({
+      name: { type: 'string', value: name },
+      states: nbt.comp(states)
+    })
+    const s = nbt.writeUncompressed(tag, 'little').toString()
+    return fnv1a32(s)
+  }
```

Comment:
As a static method:
```js
if (registry.supportFeatures('blockHashes')) {
  Block.getHash = function (name, states) {
    return getHashValue()
  }
}
```
inside the Block instance constructor:
```js
if (registry.supportFeature('blockHashes')) {
  this.hash = Block.getHash(this.name, this._properties)
}
```

Reply:
Where should first edit be made?

Reply:
I think i get it now, ill try

Reply:
> As a static method:
> 
> ```js
> 
> static getHash (name, states) {
> 
>   if (registry.supportFeature('blockHashes')) {
> 
>     return ... compute hash ...
> 
>   }
> 
> }
> 
> ```
> 
> inside the Block instance constructor:
> 
> ```js
> 
> if (registry.supportFeature('blockHashes')) {
> 
>   this.hash = Block.getHash(this.name, this._properties)
> 
> }
> 
> ```

Should supportFeature should be used as that "variable" that sets on start_game? Or other variable is still needed and supportFeature("blockHashes") will be true for mcbe >1.19.80?

Reply:
No, supportFeatures data should be inside minecraft-data features.json. It is used for handling features available in different minecraft versions

Reply:
Got it